jib-cli/src/main/java/com/google/cloud/tools/jib/cli/buildfile/BuildFileSpec.java

```diff
@@ -124,14 +128,13 @@ public BuildFileSpec(
     if (format != null) {
       this.format = ImageFormat.valueOf(format);
     }
-    this.environment = environment;
-    this.labels = labels;
-    if (volumes != null) {
-      this.volumes = volumes.stream().map(AbsoluteUnixPath::get).collect(Collectors.toSet());
-    }
-    if (exposedPorts != null) {
-      this.exposedPorts = Ports.parse(exposedPorts);
-    }
+    this.environment = (environment == null) ? ImmutableMap.of() : environment;
```

Comment:
Use of `ImmutableMap` makes me wonder if it is possible to make all the fields of the class `final`? (Although using `final` on all the fields doesn't make this class an immutable one, but there is no reason not to do so if possible.) Or, is it not feasible due to JSON parsing?

Reply:
No final should be on it. I guess I just never got around to doing it.